File: scala/src/main/scala/com/thoughtworks/jcprogram/functional/exercises/IntegerReporter.scala
package com.thoughtworks.jcprogram.functional.exercises

// Given a list of numbers like {1, 9, 4, 16, 4}
// Print the square root of all of the numbers larger than 4.
// For example, "3, 4"

object IntegerReporter {
  def main(args: Array[String]) {
    val numbers = Seq(1, 9, 4, 16, 4)

    numbers.foreach(println)
  }
}

File: java8/src/main/java/com/thoughtworks/jcprogram/functional/exercise/IntegerReporter.java
package com.thoughtworks.jcprogram.functional.exercise;

import java.util.List;

import static java.util.Arrays.asList;

// Given a list of numbers like {1, 9, 4, 16, 4}
// Print the square root of all of the numbers larger than 4.
// For example, "3, 4"

public class IntegerReporter {
    public static void main(String[] args) {
        List<Integer> numbers = asList(1, 9, 4, 16, 4);

        System.out.println(new IntegerReporter().reportSquareRootsOfLargeNumbers(numbers));

    }

    public String reportSquareRootsOfLargeNumbers(List<Integer> numbers) {
        return "";
    }

}
